examples/Makefile

```diff
@@ -1,13 +1,40 @@
 # KallistiOS ##version##
 #
-# examples/Makefile
-# (c)2001 Megan Potter
+# examples/dreamcast/Makefile
+# Copyright (C) 2003 Megan Potter
+# Copyright (C) 2024 Andy Barajas
+# Copyright (C) 2024 Paul Cercueil
 #
 
-all:
-	$(KOS_MAKE) -C $(KOS_ARCH)
+# Get a list of the sub-directories that contain a Makefile
+get_subdirs=$(foreach each,$(wildcard $(1)/*),$(if $(wildcard $(each)/Makefile),$(each),) $(call get_subdirs,$(each)))
+
+DIRS := $(call get_subdirs, $(shell pwd))
+
+.PHONY: all $(DIRS)
+
+all: $(DIRS)
+	@if [ -f errors.txt ]; then \
+		echo "\n-------------------------------------------------"; \
+		echo "$$(cat error_count.txt 2>/dev/null || echo 0) error(s) occurred during the build process:"; \
+		cat errors.txt; \
+	fi;
+	@rm -f errors.txt error_count.txt
+
+$(DIRS):
+	$(MAKE) -C $@ ; \
```

Comment:
While compiling examples I was getting this output:
```
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C /opt/toolchains/dc/kos/examples/dreamcast/sound/multi-stream ; \
	rv=$? ; \
	if [ "$rv" -ne 0 ]; then \
		echo "dreamcast/sound/multi-stream: Build failed with return code $rv" >> errors.txt; \
		echo $(($(cat error_count.txt 2>/dev/null || echo 0) + 1)) > error_count.txt; \
	fi
rm -f multistream.elf romdisk.o main.o
rm -f romdisk.o romdisk.img
/opt/toolchains/dc/kos/utils/genromfs/genromfs -f romdisk.img -d romdisk -v -x .svn -x .keepme
0    rom 1717019680       [0xffffffff, 0xffffffff] 0177777, sz     0, at 0x0     
1    .                    [0x1000008 , 0x33ea9eb ] 0040755, sz     0, at 0x20    
1    ..                   [0x1000008 , 0x33ea9e7 ] 0040755, sz     0, at 0x40     [link to 0x20    ]
1    brushing.wav         [0x1000008 , 0x33ea9ec ] 0100644, sz 1508444, at 0x60    
1    faucet.wav           [0x1000008 , 0x33ea9ed ] 0100644, sz 652380, at 0x1704e0
/opt/toolchains/dc/kos/utils/bin2o/bin2o romdisk.img romdisk romdisk_tmp.o
/opt/toolchains/dc/sh-elf/bin/sh-elf-gcc -o romdisk.o -r romdisk_tmp.o -L/opt/toolchains/dc/kos/lib/dreamcast -L/opt/toolchains/dc/kos/addons/lib/dreamcast -L/opt/toolchains/dc/kos-ports/lib -Wl,--whole-archive -lromdiskbase
rm romdisk_tmp.o
kos-cc  -c main.c -o main.o
kos-cc -o multistream.elf romdisk.o main.o   -lwav
```
Using the suggestion I was able to hide the portion of the makefile script from being printed out for each example.
```suggestion
	@$(MAKE) -C $@ ; \
```